utils/gen-styleguide-prod-config.js

```diff
@@ -0,0 +1,188 @@
+        }
+    });
+    // Let's specify the production server port.
+    lines.push(`    serverPort: 6061,`);
+    lines.push("};\n");
+
+    const data = lines.join("\n");
+    fs.writeFileSync(filepath, data, "utf8");
+}
+
+/**
+ * This function recurses through the sections and sub-sections, removing
+ * any that are marked as private. It returns the modified array with the
+ * private sections removed.
+ */
+function removePrivateSections(sections) {
+    if (!sections) {
+        return sections;
+    }
+    for (let i = sections.length; i > 0; i--) {
+        const sectionIndex = i - 1;
+        if (sections[sectionIndex].private) {
+            // eslint-disable-next-line no-console
+            console.log(`Removing private section ${sections[sectionIndex].name}`);
+
+            // If the section is private, we presume all content of the section is
+            // private too.
+            sections.splice(sectionIndex, 1);
+        } else {
+            // And now, we move into the chilren of the section and repeat.
+            sections[sectionIndex].sections =
+                removePrivateSections(sections[sectionIndex].sections);
+        }
+    }
+    return sections;
+}
+
+function maybeGetPackagePathForSection(section) {
+    const getPackageDirFromPath = filepath => {
+        if (!filepath) {
+            return null;
+        }
+
+        do {
+            const parentPath = path.dirname(filepath);
+            if (parentPath) {
+                const parentName = path.basename(parentPath);
+                if (parentName === "packages") {
+                    return filepath;
+                } else if (parentName === parentPath) {
+                    break;
+                } else {
+                    filepath = parentPath;
+                }
+            }
+        } while (filepath);
+
+        return null;
+    }
+
+    if (section.content) {
+        return getPackageDirFromPath(section.content);
+    } else {
+        // Other option is to get one of the component file paths and use that.
+        const files = getComponentFilesFromSection(section);
+        if (files.length) {
+            return getPackageDirFromPath(files[0]);
+        };
+    }
+
+    // We get here, we got nothing.
+    return null;
+}
+
+function getVersionInfoForSection(section) {
+    const pathForSection = maybeGetPackagePathForSection(section);
+    if (!pathForSection) {
+        // We can't help you.
+        return null;
+    }
+
+    const pkgJsonPath = path.join(path.resolve(pathForSection), "package.json");
+    if (fs.existsSync(pkgJsonPath)) {
+        // We have a package file, let's read it and find out what we can.
+        const pkgJson = require(pkgJsonPath);
+
+        return {
+            name: pkgJson.name,
+            lastRelease: pkgJson.version,
+            design: pkgJson.design,
+            prototype: pkgJson.private,
+        };
+    } else {
+        return null;
+    }
+}
+
+function genStyleguideProdConfig() {
```

Comment:
This is the function that orchestrates the work.

Reply:
This doesn't really need to be a function, does it? It can just operate at the top-level. Might make things a bit clearer.

Reply:
I find lots of code outside of a function to be a little strange and so wrapped it for clarity, but in the end, either way will work.